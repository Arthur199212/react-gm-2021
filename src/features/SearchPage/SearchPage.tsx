import React from 'react'
import { Filters, Footer, SearchResults } from '@app/components'
import { Header, SearchBox } from './components'

export const SearchPage = () => (
  <>
    <div className='app-search-page content'>
      <Header />
      <main className='main'>
        <SearchBox />
        <div className='container'>
          <Filters />
          <SearchResults />
        </div>
      </main>
    </div>
    <Footer />
  </>
)
