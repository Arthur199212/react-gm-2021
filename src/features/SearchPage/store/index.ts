export * from './searchPage.selectors'
export * from './searchPage.slice'
export * from './searchPage.thunks'
