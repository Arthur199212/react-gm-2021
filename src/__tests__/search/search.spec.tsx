import React from 'react'
import SearchPage, { getServerSideProps } from '@app/pages/search/[query]'
import { render } from '@app/tests/testing-utils'
import { MOCK_MOVIES } from '@app/tests/mocks/mock-data'

describe('Search Page', () => {
  const queryParamsMock = { query: { query: 'test' } }

  it('should render properly', async () => {
    const { asFragment } = render(<SearchPage />, queryParamsMock)

    expect(asFragment()).toMatchSnapshot()
  })

  it('getServerSideProps should work properly', async () => {
    const {
      props: { preloadedState }
    } = await getServerSideProps(queryParamsMock as any)

    expect(preloadedState.search.movies).toEqual(MOCK_MOVIES)
    expect(preloadedState.search.status).toEqual('success')
  })
})
