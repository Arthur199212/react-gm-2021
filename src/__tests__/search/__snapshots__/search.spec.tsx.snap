// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page should render properly 1`] = `
<DocumentFragment>
  <div
    class="app-search-page content"
  >
    <header
      class="header"
    >
      <div
        class="header-container container"
      >
        <span
          class="title"
        >
          MovieSearchApp
        </span>
        <a
          class="app-button"
          data-testid="header-button"
        >
          ADD MOVIE
        </a>
        <div
          class="app-modal-container"
          data-testid="modal-container"
        >
          <div
            class="app-modal"
          >
            <div
              class="app-movie-form-container"
              data-testid="add-movie-form-container"
            >
              <h5
                class="app-movie-form-header"
              >
                ADD MOVIE
              </h5>
              <form
                action="#"
              >
                <i
                  class="app-close-icon fas fa-times topRight"
                  data-testid="add-movie-form-close-icon"
                />
                <div
                  class="app-form-field"
                >
                  <label
                    class="app-form-field-label"
                  >
                    Title
                  </label>
                  <input
                    aria-label="title"
                    autocomplete="off"
                    class="app-input-box"
                    name="title"
                    placeholder="Title"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="app-form-field"
                >
                  <label
                    class="app-form-field-label"
                  >
                    Release date
                  </label>
                  <input
                    aria-label="release-date"
                    autocomplete="off"
                    class="app-input-box"
                    name="release_date"
                    type="date"
                    value=""
                  />
                </div>
                <div
                  class="app-form-field"
                >
                  <label
                    class="app-form-field-label"
                  >
                    Movie URL
                  </label>
                  <input
                    aria-label="movie-url"
                    autocomplete="off"
                    class="app-input-box"
                    name="poster_path"
                    placeholder="Movie Poster URL"
                    type="url"
                    value=""
                  />
                </div>
                <div
                  class="app-form-field"
                >
                  <label
                    class="app-form-field-label"
                  >
                    Genre
                  </label>
                  <input
                    aria-label="genre"
                    autocomplete="off"
                    class="app-input-box"
                    name="genres"
                    placeholder="Genge"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="app-small-modal-container"
                  data-testid="genres-form-field-small-modal"
                >
                  <div
                    class="app-small-modal"
                  >
                    <div
                      class="genres-form-field-modal-title"
                    >
                      CHOOSE GENRES
                    </div>
                    <i
                      class="app-close-icon fas fa-times topRight"
                    />
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Action"
                      >
                        Action
                        <input
                          aria-label="Action"
                          class="checkbox"
                          name="Action"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Adventure"
                      >
                        Adventure
                        <input
                          aria-label="Adventure"
                          class="checkbox"
                          name="Adventure"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Animation"
                      >
                        Animation
                        <input
                          aria-label="Animation"
                          class="checkbox"
                          name="Animation"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Comedy"
                      >
                        Comedy
                        <input
                          aria-label="Comedy"
                          class="checkbox"
                          name="Comedy"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Crime"
                      >
                        Crime
                        <input
                          aria-label="Crime"
                          class="checkbox"
                          name="Crime"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Disaster"
                      >
                        Disaster
                        <input
                          aria-label="Disaster"
                          class="checkbox"
                          name="Disaster"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Documentary"
                      >
                        Documentary
                        <input
                          aria-label="Documentary"
                          class="checkbox"
                          name="Documentary"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Drama"
                      >
                        Drama
                        <input
                          aria-label="Drama"
                          class="checkbox"
                          name="Drama"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Fantasy"
                      >
                        Fantasy
                        <input
                          aria-label="Fantasy"
                          class="checkbox"
                          name="Fantasy"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Family"
                      >
                        Family
                        <input
                          aria-label="Family"
                          class="checkbox"
                          name="Family"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="History"
                      >
                        History
                        <input
                          aria-label="History"
                          class="checkbox"
                          name="History"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Horror"
                      >
                        Horror
                        <input
                          aria-label="Horror"
                          class="checkbox"
                          name="Horror"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Music"
                      >
                        Music
                        <input
                          aria-label="Music"
                          class="checkbox"
                          name="Music"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Mystery"
                      >
                        Mystery
                        <input
                          aria-label="Mystery"
                          class="checkbox"
                          name="Mystery"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Psychological"
                      >
                        Psychological
                        <input
                          aria-label="Psychological"
                          class="checkbox"
                          name="Psychological"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Romance"
                      >
                        Romance
                        <input
                          aria-label="Romance"
                          class="checkbox"
                          name="Romance"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Satire"
                      >
                        Satire
                        <input
                          aria-label="Satire"
                          class="checkbox"
                          name="Satire"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Science Fiction"
                      >
                        Science Fiction
                        <input
                          aria-label="Science Fiction"
                          class="checkbox"
                          name="Science Fiction"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Thriller"
                      >
                        Thriller
                        <input
                          aria-label="Thriller"
                          class="checkbox"
                          name="Thriller"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="War"
                      >
                        War
                        <input
                          aria-label="War"
                          class="checkbox"
                          name="War"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                    <div
                      class="app-checkbox-form-field"
                    >
                      <label
                        class="label"
                        for="Western"
                      >
                        Western
                        <input
                          aria-label="Western"
                          class="checkbox"
                          name="Western"
                          type="checkbox"
                        />
                        <span
                          class="checkmark"
                        />
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  class="app-textarea-form-field"
                >
                  <label
                    class="app-textarea-form-field-label"
                  >
                    Overview
                  </label>
                  <textarea
                    aria-label="overview"
                    autocomplete="off"
                    class="app-textarea no-resize"
                    name="overview"
                    placeholder="Overview"
                    rows="4"
                  />
                </div>
                <div
                  class="app-form-field"
                >
                  <label
                    class="app-form-field-label"
                  >
                    Runtime
                  </label>
                  <input
                    aria-label="runtime"
                    autocomplete="off"
                    class="app-input-box"
                    name="runtime"
                    placeholder="Runtime"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-actions-container"
                >
                  <a
                    aria-label="reset"
                    class="app-button"
                  >
                    RESET
                  </a>
                  <a
                    aria-label="submit"
                    class="app-button"
                  >
                    SUBMIT
                  </a>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main
      class="main"
    >
      <div
        class="search-box-container container"
      >
        <div
          class="search-header"
        >
          <h1
            class="title"
          >
            FIND YOUR MOVIE
          </h1>
        </div>
        <div
          class="search-container"
        >
          <input
            autocomplete="off"
            class="app-input-box"
            placeholder="What do you want to watch?"
            type="text"
            value=""
          />
          <a
            class="app-button"
          >
            SEARCH
          </a>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="filters-container"
          data-testid="filters-container"
        >
          <ul
            class="app-tabs"
          >
            <li
              class="app-tab active"
            >
              all
            </li>
            <li
              class="app-tab"
            >
              comedy
            </li>
            <li
              class="app-tab"
            >
              crime
            </li>
            <li
              class="app-tab"
            >
              documentary
            </li>
            <li
              class="app-tab"
            >
              horror
            </li>
          </ul>
          <div
            class="dropdown-container"
          >
            <span
              class="label"
            >
              SORT BY
            </span>
            <div
              class="dropdown-menu"
            >
              <a
                class="app-button"
                data-testid="filters-dropdown-button"
              >
                release date
              </a>
              <ul
                class="app-dropdown"
                data-testid="filters-dropdown"
              >
                <li
                  class="app-dropdown-item"
                  data-testid="dropdown-item"
                >
                  rating
                </li>
                <li
                  class="app-dropdown-item"
                  data-testid="dropdown-item"
                >
                  release date
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="search-results-container centered"
        >
          <div
            class="app-dual-ring-spinner"
          />
        </div>
      </div>
    </main>
  </div>
  <footer
    class="app-footer"
  >
    MovieSearchApp
  </footer>
</DocumentFragment>
`;
